src/cr/cube/dimension.py

```diff
@@ -725,6 +734,214 @@ def _elements(self):
+        replaced with the alias for subvariables in the `._shimmed_dimension_dict`.
+        """
+        return tuple(
+            element["id"] for element in self._dimension_dict["type"]["elements"]
+        )
+
+    def _replaced_element_transforms(self, element_transforms):
+        """Replace the dictionary keys of element transforms with aliases
+
+        The element transforms identify which element they refer to by their key in
+        the element_transforms object. Before it is shimmed, this can identify them
+        in many different ways. The shim replaces these with the alias.
+        """
+        # --- The name of the element transform object is a string. This string is
+        # --- assumed to refer to the subvariable id, unless there is a "key", which
+        # --- can be "alias", in which case it refers to the alias, or if no
+        # --- subvariable is found with the id, in which case it's assumed to be the
+        # --- element_id
+        # --- TODO: This logic about "key" is not supported by the validation
+        # --- on the deck schema, so maybe we should remove it?
+        key = element_transforms.get("key")
+        if key == "alias":
+            # --- Already keyed by alias, no changes needed
+            return element_transforms
+
+        old_keys = tuple(element_transforms.keys())
+
+        if key == "subvar_id":
+            # --- translate from subvariable id
+            new_keys = tuple(
+                self._subvar_aliases[self._subvar_ids.index(_id)]
+                if _id in self._subvar_ids
+                else None
+                for _id in old_keys
+            )
+        else:
+            # --- Otherwise use usual translation logic
+            new_keys = tuple(self.translate_element_id(_id) for _id in old_keys)
+
+        return {
+            nkey: element_transforms[old_keys[i]]
+            for i, nkey in enumerate(new_keys)
+            if nkey is not None
+        }
+
+    def _replaced_order_element_ids(self, element_ids):
+        """Replace the list list of element ids with a list of aliases
+
+        The explicit order transform includes a list of ids that can be specified in
+        many different ways, this translate them to the subvariable aliases.
+        """
+        return [self.translate_element_id(_id) for _id in element_ids]
+
+    @lazyproperty
+    def _subvar_aliases(self):
+        """Tuple of string aliases for each element of a subvariable dimension
+
+        Fall back to the `element_id` if the alias doesn't exist (this happens in one
+        fixture, but I don't think it can happen in the real world anymore.)
+        """
+        return tuple(
+            element.get("value", {}).get("references", {}).get("alias", element["id"])
+            for element in self._dimension_dict["type"]["elements"]
+        )
+
+    @lazyproperty
+    def _subvar_ids(self):
+        """Tuple of string subvariable ids for each element of a subvariable dimension"""
```

Comment:
same on this one.

Also we should note in the docstring that referring to this property is only valid on a subvars dimension. This will raise `KeyError` on a CAT dimension I'm pretty sure.